### PR TITLE
JUJU-7125/add-jimm-facade-for-role

### PR DESCRIPTION
**Pull request by SimoneDutto**
## Description

Add facade methods for roles.
The actual implementations are empty.

Fixes _JIRA/GitHub issue number_

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [ ] Covered by unit tests
- [ ] Covered by integration tests

### CODE REVIEW

File: internal/jujuapi/role.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	const op = errors.Op("jujuapi.RemoveRole")
+
+	if err := r.jimm.RemoveRole(ctx, r.user, req.Name); err != nil {
+		zapctx.Error(ctx, "failed to remove role", zaputil.Error(err))
+		return errors.E(op, err)
+	}
+	return nil
+}
+
+// ListRole lists relational access control roles within JIMMs DB.
```
Comment (kian99) [inline internal/jujuapi/role.go]: Godoc is a bit weird, why not just,
```suggestion
// ListRole lists access control roles within JIMMs DB.
```

File: pkg/api/params/params.go
Comment (kian99) [inline pkg/api/params/params.go:1]: All the docstrings in this file still mentioned `group` at the end of the sentence.

File: internal/jimm/role.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	"github.com/canonical/jimm/v3/internal/common/pagination"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/openfga"
+)
+
+// AddRole creates a role within JIMMs DB for reference by OpenFGA.
+func (j *JIMM) AddRole(ctx context.Context, user *openfga.User, name string) (*dbmodel.RoleEntry, error) {
+	return &dbmodel.RoleEntry{}, nil
+}
```
Comment (kian99) [inline internal/jimm/role.go:15]: I guess you still need to hook all of these up to the DB methods. Will that be for this PR or a follow-up?

Comment (SimoneDutto) [reply internal/jimm/role.go:15]: yes, the jimm implementation is for a follow-up.
So this 300lines skeleton pr is not distracting from the actual implementation.

**Review by kian99 — APPROVED** [review]
Looks fine but you might want to start from the bottom up, adding the `internal/jimm` logic first with tests then you can add the facade layer with tests. Because as it stands this PR doesn't have tests so I'm not sure if it should go in even though it's just skeleton.
I'll approve as I think it's fine this time, but worth thinking about.

**Comment by SimoneDutto** [comment]
> Looks fine but you might want to start from the bottom up, adding the `internal/jimm` logic first with tests then you can add the facade layer with tests. Because as it stands this PR doesn't have tests so I'm not sure if it should go in even though it's just skeleton. I'll approve as I think it's fine this time, but worth thinking about.

yes, i did it just so because I was waiting on other prs to land before I could work on jimm's implementations.
Maybe an unnecessary optimization

**Comment by SimoneDutto** [comment]
closing because @kian99 is right, it was an unnecessary PR. It is better to wait or doing the other way around it

**Pull request closed without merging**